src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -43,21 +42,23 @@ protected override ConversionsBase WithNullabilityCore(bool includeNullability)
         public override Conversion GetMethodGroupDelegateConversion(BoundMethodGroup source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
             // Must be a bona fide delegate type, not an expression tree type.
-            if (!destination.IsDelegateType())
+            if (!(destination.IsDelegateType() || destination.SpecialType == SpecialType.System_Delegate))
             {
                 return Conversion.NoConversion;
             }
 
-            var (methodSymbol, isFunctionPointer, callingConventionInfo) = GetDelegateInvokeOrFunctionPointerMethodIfAvailable(destination);
+            var (methodSymbol, isFunctionPointer, callingConventionInfo) = GetDelegateInvokeOrFunctionPointerMethodIfAvailable(source, destination, ref useSiteInfo);
             if ((object)methodSymbol == null)
             {
                 return Conversion.NoConversion;
             }
 
+            Debug.Assert(destination.SpecialType == SpecialType.System_Delegate || methodSymbol == ((NamedTypeSymbol)destination).DelegateInvokeMethod);
```

Comment:
>methodSymbol == ((NamedTypeSymbol)destination).DelegateInvokeMethod [](start = 83, length = 67)

This doesn't seem correct. Function pointer methods aren't going to have a DelegateInvokeMethod.

Reply:
Nevermind, this is the helper that should only be receiving delegate type requests.

---
In reply to: [618602982](https://github.com/dotnet/roslyn/pull/52448#discussion_r618602982) [](ancestors = 618602982)